plugins/destination/postgresql/client/write.go

```diff
@@ -39,12 +48,13 @@ func (c *Client) Write(ctx context.Context, tables schema.Tables, res <-chan *pl
 		if batch.Len() >= c.batchSize {
 			br := c.conn.SendBatch(ctx, batch)
 			if err := br.Close(); err != nil {
-				if _, ok := err.(*pgconn.PgError); !ok {
+				var pgErr *pgconn.PgError
+				if !errors.As(err, &pgErr) {
 					// not recoverable error
 					return fmt.Errorf("failed to execute batch: %w", err)
 				}
 				atomic.AddUint64(&c.metrics.Errors, 1)
-				c.logger.Error().Err(err).Msgf("failed to execute batch with pgerror")
+				c.logger.Error().Err(&pgErrorWrap{PgError: pgErr}).Msg("failed to execute batch with pgerror")
```

Comment:
```suggestion
				c.logger.Error().Err(err).Str("table", table.Name).Msg("failed to execute batch with pgerror")
```

Would something like this not achieve the goal of adding the table name to the error log?

Reply:
I don't think so, as we only send the batch if `batch.Len() >= c.batchSize` so if we take the table name from the loop it will print the last table and not necessarily the one generated the error

Reply:
Ah okay I see - but yeah having it as a log field sounds good 👍

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/5552/commits/0eecc4a68965feb8ab217d6788d2548c08772056